sound/hybrid_font.h

```diff
@@ -138,24 +140,57 @@ class HybridFont : public SaberBase {
     }
   }
   
-  void StartSwing() override {
+  void StartSwing(const Vec3& gyro, float swingThreshold_, float slashThreshold_) override {
+    float speed = sqrtf(gyro.z * gyro.z + gyro.y * gyro.y);
+    uint32_t now = micros();
+    uint32_t delta_micros = now - last_swing_micros_;
+    last_swing_micros_ = now;
+    float delta = delta_micros * 0.000001;
+    float current_acceleration = (speed - last_speed_) / delta;
+    last_speed_ = speed;
+    float filter_factor = powf(0.01, delta);
+    swing_acceleration_ = swing_acceleration_ * filter_factor + current_acceleration * (1 - filter_factor);
+    if (swing_acceleration_ > slashThreshold_ && slsh.files_found()) {
+      doSlash_ = true;
+      swinging_ = true;
+    }
     if (!guess_monophonic_) {
-      if (swing_player_) {
+      if (swing_player_ && swinging_) {
         // avoid overlapping swings, based on value set in ProffieOSSwingOverlap.  Value is
         // between 0 (full overlap) and 1.0 (no overlap)
         if (swing_player_->pos() / swing_player_->length() >= config_.ProffieOSSwingOverlap) {
           swing_player_->set_fade_time(swing_player_->length() - swing_player_->pos());
           swing_player_->FadeAndStop();
           swing_player_.Free();
-          swing_player_ = PlayPolyphonic(&swng);
         }
       }
-      else if (!swing_player_) {
-        swing_player_ = PlayPolyphonic(&swng);
+      if (!swing_player_) {
+        if (doSlash_) {
```

Comment:
This doesn't seem to work the way I imagined it would.
The whole point of using the acceleration was to have one start point (based on speed) and at that point we would decide if it's a slash or a swing based on the acceleration.

This code seems to play a slash as soon as the acceleration is high enough, which could happen before or after the swing gets kicked off. It seems like the swing and slash would fight each other quite a bit this way....

Reply:
You're right of course.  I've corrected this error and the slashes are very predictable now.